### PR TITLE
Disable settings editor override

### LINKED ISSUE #117203: Settings editor is partially unusable when configuring a custom editor for JSON
Steps to Reproduce:

1. configure workbench editor associations to open *.json with a custom editor (e.g. HEX)
2. try to change this setting later

=> 🐛 you cannot because in order to edit the values, you need to go to the JSON but it will open the custom editor always:

image

Imho we should have a way to prevent the override from happening fro built in files such as settings.

Only workaround is to entirely reset the setting.

//cc @lramos15

### PR DESCRIPTION
**Pull request by lramos15**
This PR fixes #117203

Adds the disable override flag to settings and keybindings editors. This allows them to still work even if an editor association is set for that specific file type.

### CODE REVIEW

File: src/vs/workbench/browser/editor.ts [hunk 1]
```diff
@@ -174,8 +194,137 @@ class EditorRegistry implements IEditorRegistry {
...
+		};
+	}
+
+	private updateEditorAssociationsSchema() {
+		const enumValues: string[] = [];
+		const enumDescriptions: string[] = [];
+
+		const editorTypes: IEditorType[] = [DEFAULT_EDITOR_ASSOCIATION];
+
+		for (const [, handler] of this.editorTypesHandlers) {
```
Comment (rheh) [inline src/vs/workbench/browser/editor.ts:311]: nit-pick: this looks like a array map would be better suited? e.g.

```
const editorTypes: IEditorType[] = this.editorTypesHandlers.map(([, handler] => ...handler.getEditorTypes());
```

**Review by rheh — APPROVED** [review]
One tiny nit-pick.  Code look solid to me.

**Comment by roblourens** [comment]
The last commit looks good to me, but please fix the PR to show the right commits and changes before merging.

**Comment by roblourens** [comment]
Here's a command to fix your PR:

```
 git rebase --onto origin/main lramos15/editorServiceOpenWith
```

then `git push --force`

Sometimes before I do a rebase, I run something like `git checkout -b tmp`, then `git checkout <original branch>`. This ensures that if I do the rebase wrong, all my original commits will still exist in the `tmp` branch.

Read the `git help rebase` section on rebase for changing the branch that the "topic" branch is based off, from "next" to "master". I have to check that every time to get the rebase arguments right.

**Comment by roblourens** [comment]
I suspect that `git merge origin/main` would also fix it, you can try if you want, but I think the history in a case like this will be much cleaner if you do a rebase.

**Comment by lramos15** [comment]
I had to remake the PR as I was unable to fix the history closing this one

**Pull request closed without merging**